### DISCUSSION TITLE
Cutom Actions v4 for Class library(.Net Framework 4.7.2)

### DISCUSSION THREAD

**Discussion by malyovanik** [wixtoolset/issues#7776] [Questions] [answered]
Hello everyone, 
Could you help me resolve a problem with connection custom actions for Class Library(.Net Framework 4.7.2) project?
I just created new project class library, and after adding reference to custom actions using Nuget packages, I have errors : 
1 )
""..\packages\WixToolset.Dtf.CustomAction.4.0.2\build\..\tools\WixToolset.Dtf.MakeSfxCA.exe" "..\ClassLibrary1\obj\Debug\ClassLibrary1.CA.dll" "..\packages\WixToolset.Dtf.CustomAction.4.0.2\build\..\tools\x86\SfxCA.dll" "..\ClassLibrary1\obj\Debug\ClassLibrary1.dll" "@..\ClassLibrary1\obj\Debug\ClassLibrary1.CA.rsp"" exited with code 1.
2) 
Invalid argument: No CA or UI entry points found in module: 
..\ClassLibrary1\obj\Debug\ClassLibrary1.dll	
..\SystoLock\ClassLibrary1\EXEC

**Comment by malyovanik** [comment, marked as answer]
https://github.com/oleg-shilo/wixsharp/issues/262

**Reply by robmen** [reply]
If you use the "Custom Action for WiX v4" template in [HeatWave](https://www.firegiant.com/docs/heatwave/), it will set you up for success right away.
